### DISCUSSION TITLE
Sorting by ASC. shows null or empty values first (MongoDB)

### DISCUSSION THREAD

**Discussion by mTrCarlos** [strapi/strapi#6615] [Legacy Topics]
As the title says, entries with null or undefined values are being sorted first before filled ones in ascendant mode. It seems it comes all the way from mongo side and how they handle different type values. From mongodb docs:
*null has much lower canonical value than that of number/string*

As a logical though I understand why is it like this, but from a user perspective and expectations I think its better to sort those entries as last ones. 
So I coded a service trying to fix this up, as currently we only need one collection to behave like that. I found a couple solutions that required an aggregation which I could easy pass it through model.aggregate method. Something along these lines:

```
db.col.aggregate([
    { $addFields: { fieldType: { $type: "$whateverSortingField" } } },
    { $sort: { fieldType: 1, whateverSortingField: 1 } },
    { $project: { "fieldType": 0 } }
])
```

or 

*not tested yet*
```
db.col.aggregate([
    { $project: { whateverSortingField: { $ifNull: [ "$amount", -1*(<mimimum value>)* ] } } },
    { $sort : { amount : (-1 or 1 depending on the order you want) }}
])
```

The problem was that all the relations were missing, as this method its from Moongose and has nothing to do with Strapi. I tried to look into Strapi code for some method that could populate the relations or let me do an aggregation, but I just found the *buildQuery* method from mongoose connector, which accepts an aggregate boolean parameter but only a $where block (filters.where if I'm right). I'm sure I missed something so I'll keep looking into it.

So my last option right now its to populate each relation fields on the service code, which are a lot on this case, and may need constant maintenance when Strapi code its update nor we add new relations to the collection, so I don't like it at all.
Is there any alternative way/method that could work for this ? Maybe there's something on mongo configuration I can tweak ? Also, Shouldn't Strapi by default had something for this ?

Thank you!

**Comment by souljorje** [comment]
+1
